homeassistant/components/light/tikteck.py

```diff
@@ -0,0 +1,146 @@
+        if light.is_valid:
+            lights.append(light)
+
+    add_devices(lights)
+
+
+class TikteckLight(Light):
+    """Representation of a Tikteck light."""
+
+    def __init__(self, device):
+        """Initialize the light."""
+        import tikteck
+
+        self._name = device['name']
+        self._address = device['address']
+        self._password = device['password']
+        self._brightness = 255
+        self._rgb = [255, 255, 255]
+        self._state = False
+        self.is_valid = True
+        try:
+            self._bulb = tikteck.tikteck(self._address, "Smart Light",
+                                         self._password)
+            self._bulb.connect()
+        except Exception as e:
+            self.is_valid = False
+            _LOGGER.error(
+                "Failed to connect to bulb %s, %s (%s)", self._address,
+                self._name, e)
+
+    @property
+    def unique_id(self):
+        """Return the ID of this light."""
+        return "{}.{}".format(self.__class__, self._address)
+
+    @property
+    def name(self):
+        """Return the name of the device if any."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return true if device is on."""
+        return self._state
+
+    @property
+    def brightness(self):
+        """Return the brightness of this light between 0..255."""
+        return self._brightness
+
+    @property
+    def rgb_color(self):
+        """Return the color property."""
+        return self._rgb
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_TIKTECK_LED
+
+    def set_state(self, red, green, blue, brightness):
+        """Set the bulb state."""
+        initial = time.time()
+        # BLE connections may not be stable. If we receive an error, spend up
+        # to 10 seconds attempting to reconnect before giving up.
+        while True:
+            if time.time() - initial >= 10:
+                return False
+            try:
+                self._bulb.set_state(red, green, blue, brightness)
+                return True
+            except Exception as e:
+                _LOGGER.error("Error: %s", e)
+                try:
+                    self._bulb.connect()
+                except Exception:
+                    pass
+
+    def turn_on(self, **kwargs):
+        """Turn the specified light on."""
+        self._state = True
+
+        rgb = kwargs.get(ATTR_RGB_COLOR)
+        brightness = kwargs.get(ATTR_BRIGHTNESS)
+
+        if rgb is not None:
+            self._rgb = rgb
+        if brightness is not None:
+            self._brightness = brightness
+
+        self.set_state(self._rgb[0], self._rgb[1], self._rgb[2],
+                       self.brightness)
+
+    def turn_off(self, **kwargs):
+        """Turn the specified light off."""
+        self._state = False
+        self.set_state(0, 0, 0, 0)
+
+    def update(self):
```

Comment:
If there is no update method, you can remove it. It will default to be a no-op. In that case you also should also:
 - Add property [`should_poll`](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/entity.py#L83) that returns `False`
 - Add property [`assumed_state`](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/entity.py#L146-L149) that returns `True`
 - Add `self.schedule_update_ha_state()` at the end of `turn_on` and `turn_off`